docs/developer-guide/working-with-rules.md

```diff
@@ -26,6 +26,8 @@ Here is the basic format of the source file for a rule:
 
 module.exports = {
     meta: {
+        type: "suggestion",
```

Comment:
In seeing all the discussion about what `type` each rule should be (and seeing how some of them don't necessarily match `docs.catetegory` as was initially noted [here](https://github.com/eslint/eslint/issues/10855#issuecomment-421090765)), it seems like we're heavily weighting what the fix for the error would entail when we're deciding these categories.

I wonder if we should make this more explicit and call it `fixType`? I like the idea of just having a `type` field, but I'm not sure it's clear by the name what criteria we're using when we decide on these categories.

I don't want to hold this up; just wanted to see what others think.

Reply:
We talked about this initially in the TSC meeting when we accepted the proposal. Though discussion about what type of fixes are applied has helped inform the type of each rule, I don't see this as just describing the type of fix that is applied (which is already specified, to some degree, by `meta.fixable`). The intent of `meta.type` is to help us better determine the importance of the rule rather than the type of fix to be applied. While this info is being used solely for fixes right now, there are a lot of ways we can use this information going forward.

Reply:
Reference: https://gitter.im/eslint/tsc-meetings?at=5bad404e01fb5f4ae13176fd